include/zephyr/net/wifi_credentials.h

```diff
@@ -56,6 +62,16 @@ struct wifi_credentials_header {
 	uint32_t timeout;                 /**< Timeout for connecting to the network */
 	uint8_t bssid[WIFI_MAC_ADDR_LEN]; /**< BSSID (Basic Service Set Identifier) */
 	uint8_t channel;                  /**< Channel on which the network operates */
+
+	/** Anonymous identifier (Limited to 16 bytes due to settings subsystem overflow) */
+	char anon_id[16];
+
+	uint8_t aid_length;               /**< Length of the Anonymous identifier */
```

Comment:
this is still old format, I would first have a commit that fixes all docstrings to be compliant to Zephyr and then do these changes.